### PR TITLE
3211 ndp references upload and select file from linksrepository

### LINKED ISSUE #3211: NDP: References -> Upload and select file from Links&Repository
** Allow users to select and upload files from Links (available in Links & Repository section)**

### PR DESCRIPTION
**Pull request by sorja**
- [x] Add [dropzone](https://www.dropzone.dev/)
- [x] Add vertical separator
- [x] Rename 'Add data sources' to 'Repository'
- [x] Allow selecting only local country files from repository
- [x] Enter the link without new line (text is appended after cursor)
- [x] Allow possibility to upload file within the popup
- [x] Add text where cursor is
- [ ] Translations

https://github.com/openforis/fra-platform/assets/5508251/dc30e34f-5864-47fc-b56f-f95fbf6ee942

### CODE REVIEW

File: src/client/pages/OriginalDataPoint/components/DataSources/References/AddFromRepository/AddFromRepository.tsx [hunk 1]
```diff
@@ -0,0 +1,153 @@
+import React, { useState } from 'react'
```
Comment (sorja) [inline src/client/pages/OriginalDataPoint/components/DataSources/References/AddFromRepository/AddFromRepository.tsx:1]: This is new

File: package.json [hunk 2]
```diff
@@ -255,7 +257,8 @@
     "unified": "^10.1.2",
     "uuid": "^8.3.2",
     "webpack-node-externals": "^3.0.0",
-    "winston": "^3.8.2"
+    "winston": "^3.8.2",
+    "yarn": "^1.22.19"
```
Comment (minotogna) [inline package.json:261]: why yarn ? and add ?

File: src/client/components/ButtonCheckBox/ButtonCheckBox.tsx [hunk 3]
```diff
@@ -6,7 +6,7 @@ import classNames from 'classnames'
 
 type Props = {
   onClick: React.MouseEventHandler<HTMLButtonElement>
-  label: Array<string> | string
+  label: Array<string> | string | React.ReactNode
```
Comment (minotogna) [inline src/client/components/ButtonCheckBox/ButtonCheckBox.tsx:9]: too many types here... can't we use just one ReactNode ?

File: src/client/components/Dropzone/Dropzone.tsx [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+import { useDropzone } from 'react-dropzone'
+import { useTranslation } from 'react-i18next'
+
+import classNames from 'classnames'
+
+type Props = {
+  onDrop: (files: Array<File>) => void
+}
+
+const Dropzone: React.FC<Props> = (props: Props) => {
```
Comment (minotogna) [inline src/client/components/Dropzone/Dropzone.tsx:12]: let's call this FileDrop instead ? we can mask the drop zone implementation

File: src/client/pages/CountryHome/FraHome/Links/Links.tsx [hunk 5]
```diff
@@ -1,26 +1,21 @@
...
-import { CountryIso } from 'meta/area'
 import { Authorizer, Users } from 'meta/user'
 
-import { useAppDispatch } from 'client/store'
 import { useCountry } from 'client/store/area'
 import { useAssessment, useCycle } from 'client/store/assessment'
-import { AssessmentFilesActions } from 'client/store/ui/assessmentFiles'
-import { useAssessmentFiles } from 'client/store/ui/assessmentFiles/hooks'
+import { useGetAssessmentFiles, useUpdateAssessmentFiles } from 'client/store/ui/assessmentFiles'
+import { useAssessmentFiles, useDeleteAssessmentFile } from 'client/store/ui/assessmentFiles/hooks'
```
Comment (minotogna) [inline src/client/pages/CountryHome/FraHome/Links/Links.tsx:13]: 1. they should all imported from client/store/ui/assessmentFiles 
2. should we move them under cycle data ? or not ? what do you think?

Comment (sorja) [reply src/client/pages/CountryHome/FraHome/Links/Links.tsx:13]: sure

**Pull request closed without merging**